components/core/src/clp/ffi/ir_stream/Serializer.cpp

```diff
@@ -59,6 +258,207 @@ auto Serializer<encoded_variable_t>::change_utc_offset(UtcOffset utc_offset) ->
     serialize_utc_offset_change(m_curr_utc_offset, m_ir_buf);
 }
 
+template <typename encoded_variable_t>
+auto Serializer<encoded_variable_t>::serialize_msgpack_map(msgpack::object const& msgpack_map
+) -> bool {
+    if (msgpack::type::MAP != msgpack_map.type) {
+        return false;
+    }
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+    auto const& map{msgpack_map.via.map};
+    if (0 == map.size) {
+        serialize_empty_object(m_ir_buf);
+        return true;
+    }
+
+    m_schema_tree.take_snapshot();
+    m_schema_tree_node_buf.clear();
+    m_key_group_buf.clear();
+    m_value_group_buf.clear();
+
+    bool failure{false};
+    vector<MsgpackMapNode> working_stack;
+    working_stack.emplace_back(SchemaTree::cRootId, map.ptr, static_cast<size_t>(map.size));
+    while (false == working_stack.empty()) {
+        auto& curr{working_stack.back()};
+        if (false == curr.has_next_child()) {
+            working_stack.pop_back();
+            continue;
+        }
+        auto const& [key, val]{curr.get_next_child()};
+        auto const opt_schema_tree_node_type{get_schema_tree_node_type_from_msgpack_val(val)};
+        if (false == opt_schema_tree_node_type.has_value()) {
+            failure = true;
+            break;
+        }
+
+        auto const schema_tree_node_type{opt_schema_tree_node_type.value()};
```

Comment:
```suggestion
        auto const schema_tree_node_type{opt_schema_tree_node_type.value()};

```

Reply:
It's a good idea to try and break up large blocks of code into smaller blocks so that it's easier to read.